### ISSUE TITLE
Constant Calibration keeps drifting

### ISSUE THREAD

**Issue by Activusto** [bdunderscore/OpenVR-SpaceCalibrator#3]
As title says, Constant Calibration keeps drifting
It's really annoying it's gotten to the point where having CC turned off has less issues (still a ton though).
Not really sure what I'm doing wrong here.

**Comment by Hotohori** [comment]
Yeah, that is strange, it worked for me with a Tundra Tracker on the Headset for hours, no drift at all anymore where my Index Controller with the Pico Neo 3 Link needs a recalibration after a minute without that feature.

What did you played as it drifted? Maybe you moved too less? Constant Calibration shows also a small log in the window when it is working, maybe that infos would help as well.

**Comment by Activusto** [comment]
I was playing VRChat, yet I was still moving around enough that it should have been calibrating.
I've moved the Tundra Tracker onto it's own Dongle and moved it further away from the other one (Controllers/Trackers)
So I'm hoping that'll help.

**Comment by Activusto** [comment]
Test 1.
It doesn't calibrate for ages even tho I do move.
![image](https://user-images.githubusercontent.com/15093941/185732108-6027f9e1-235b-42c9-89e1-a097bbf3b3ac.png)

**Comment by Activusto** [comment]
Slght drift (noticable when ilook up/down)
![image](https://user-images.githubusercontent.com/15093941/185735437-1393d8c9-b596-4faf-b350-7e907ade28a9.png)

**Comment by Activusto** [comment]
Another drift but i did leave the room which tends to mess up in general.
![image](https://user-images.githubusercontent.com/15093941/185736372-13ae4200-cb24-4933-9e0e-362545a75ddf.png)

**Comment by Activusto** [comment]
Another one
![image](https://user-images.githubusercontent.com/15093941/185738767-2b599688-7d67-4f94-9693-b52a61983984.png)

**Comment by Activusto** [comment]
One last 
![image](https://user-images.githubusercontent.com/15093941/185743541-f2d4a636-acfc-4cd8-99e7-28acf29bf500.png)

**Comment by bdunderscore** [comment]
Please try again with the r1 release, I've added a static correction mode that avoids large corrections when you go from sitting still to moving enough for calibration. It also adds a log and graphs to help diagnose issues.

**Comment by bdunderscore** [comment]
also - if you're building yourself, use a release build. debug builds are extremely slow and may have issues collecting sufficient data

**Comment by Activusto** [comment]
I'm not building it myself, not smart enough for that.
When does the R1 release come out?

**Comment by bdunderscore** [comment]
https://github.com/bdunderscore/OpenVR-SpaceCalibrator/releases/tag/v1.4-bd_-r1

**Comment by Activusto** [comment]
Still get massive drifting.
![image](https://user-images.githubusercontent.com/15093941/188125628-05dc7e27-40ee-40aa-ba41-6acf763d4e46.png)

**Comment by Activusto** [comment]
![image](https://user-images.githubusercontent.com/15093941/188126024-4983db7a-c8dc-486b-ac9e-f9bf40ca3a6b.png)

**Comment by Activusto** [comment]
It seems to recalibrate way more often now which in theory is great but it doesn't let the environment drift before it starts to.

**Comment by Activusto** [comment]
![image](https://user-images.githubusercontent.com/15093941/188261968-a861b1a1-bd28-42cb-a8ad-c5dcdc14c1b3.png)

**Comment by Activusto** [comment]
right after i calibrated it broke instantly
![image](https://user-images.githubusercontent.com/15093941/188262101-2bc4eaa1-5f48-470d-8f64-d5c95362253b.png)

**Comment by Activusto** [comment]
I was having too many drifting issues with R1 so I'm gonna downgrade and see how it goes.
Might just be a me issue with this, maybe today is just a shit day for tracking, or maybe it's the software, I'll find out soon enough.

**Comment by Activusto** [comment]
I had more consistency with the Original release than R1
I wonder if it has to do with Stationary Calibration, I'll try turning that off and seeing how it goes.

**Comment by Activusto** [comment]
That did not help, still drifted insanely.
On an unrelated note, can you make it so you can hide the Puck outside of Continuous Calibration?
Sometimes it takes a bit to calibrate, or sometimes I just don't use it (sometimes I get better results without it on) so I'm stuck with a Chest tracker on my Head in VRChat and it's really awkward.

**Comment by bdunderscore** [comment]
@Activusto 
I've cut a new release at https://github.com/bdunderscore/OpenVR-SpaceCalibrator/releases/tag/v1.4-bd_-r2 - try adjusting the "Recalibration threshold" on the "Alignment speeds" tab to be a bit higher (r0 was somewhere around 4, default is 1.5). Let me know if this helps.

Re hiding the target device before calibration happens - I'll think about that one, the way the code is organized it's a bit of a pain.

**Comment by bdunderscore** [comment]
I will note that I don't have access to a Reverb test environment, so if continuous calibration doesn't work for you with that hardware I might have to leave it up to someone with that equipment to try to fix it themselves.

**Comment by Activusto** [comment]
Thanks, I'll keep you posted.

**Comment by Activusto** [comment]
Still doing some tests, got a few questions tho.
Outside of the Threshold, what does the rest of the settings do under Alignment speeds?
Bit of a small brain so I don't really understand the descriptions.

Also, this is a long shot but are there plans to....optimize the program so it doesn't take as much resources?
I've noticed that it affects the performance.
If not, then that's fine, the main game I really need this in is VRChat and that game already runs like garbage.

Every other game I can get away without using Constant Calibration.

**Comment by bdunderscore** [comment]
Hmm. I haven't noticed performance issues on my side. Can you set one of the graphs to show Processing Time and get a screenshot?

**Comment by Activusto** [comment]
Yeah, next time I play I'll show it off.

**Comment by Activusto** [comment]
![image](https://user-images.githubusercontent.com/15093941/193277456-a53f4abc-d107-4205-ba3a-504160792e5f.png)
The calibration is not accurate anymore.
Whenever I turn my controllers aren't 1-1.
Are there settings I can change?

**Comment by Activusto** [comment]
Turns out it was static cal8bration, turned it off for now.
Here is the average Processing times. 
![image](https://user-images.githubusercontent.com/15093941/193279703-0fac0e43-b04c-4673-8e0f-906e537500f3.png)

**Comment by Activusto** [comment]
Maybe it's less the Calibrator that is messing with performance, but just too much going on in general, and the Calibrator is the "nail in the coffin" for performance.
Not necessarily the fault of the calibrator.

**Comment by bdunderscore** [comment]
Static calibration can result in loss of accuracy if the tracker is not firmly attached to the HMD. Perhaps that's the issue for you? It doesn't look like it's using too much CPU time based on that graph.

**Comment by bdunderscore** [comment]
I've found that the graphs are actually consuming quite a bit of CPU. If you're having issues with CPU load, try switching to the "alignment speeds" tab to avoid this. I'll reduce that overhead a bit for the next release.

**Comment by Activusto** [comment]
I'll keep that in mind, cheers.
Btw few weekends have past, it still drifts but not as intensely, and it fixes itself way quicker so by the time it goes out of place, it recalibrates and fixes it.

Been cases where I have had to manually calibrate and turn off static calibration, but in all honesty, it's just leagues better that I haven't been annoyed by it.
I don't really have any complaints, it works exactly how I want it to work now.

**Comment by Activusto** [comment]
Well it's been over a month and so far my opinion has yet to change.
Especially when I had to reinstall it due to hard drive failure, setting it back up with all my settings was daunting as I didn't have them memorised but clearly this shows that I didn't find a specific sweet spot for me, a lot of my problems are just ironed out.
